codegen: built-in primitives should emit -prefixed C symbols to avoid libc collisions

## Context

Built-in Aether primitives like `sleep(ms)` are emitted to the generated C as bare `sleep(int)`, which conflicts with libc's `unsigned int sleep(unsigned int)` whenever the user's program (or anything it imports transitively) declares an `extern sleep(...)` of its own.

Surfaced concretely while building the `std.http.client` v2 integration test (PR in flight). The test's echo server actor needs a handler that delays before responding so the client's `set_timeout` can be exercised. Adding `extern sleep(ms: int)` to the probe — to make explicit what the program uses — produces:

    /tmp/_ae_NNNNN.c:530:6: error: conflicting types for 'sleep'; have 'void(int)'
      530 | void sleep(int);
          |      ^~~~~
    /usr/include/unistd.h:464:21: note: previous declaration of 'sleep' with type 'unsigned int(unsigned int)'

Workaround in the PR: don't declare `extern sleep`, rely on the built-in being silently emitted. Works but means the dependency on `sleep` is invisible at the import level — the kind of thing a reader will trip over.

## Proposed fix

Codegen should emit Aether-side built-in primitives under an `aether_`-prefixed C symbol so they're guaranteed not to collide with libc, OS headers, or any third-party C library a user binds via `--extra`.

| Aether built-in | Today (collides with libc) | Proposed C symbol |
|---|---|---|
| `sleep(ms)` | `sleep(int)` | `aether_sleep_ms(int)` |
| `print(s)` | `print(...)` (currently doesn't collide because no libc `print` exists) | `aether_print(...)` (defensive) |
| `exit(n)` | `exit(int)` | `aether_exit(int)` (currently OK because libc's prototype matches) |
| `spawn(...)`, `send(...)` | actor primitives — already mangled internally | already-namespaced, leave alone |

Same problem class as the `pqueue_*` ↔ OpenSSL collision fixed in PR #219 — Aether stdlib symbols got the `aether_` prefix to avoid the same vendor-lib collision. Built-ins should follow the same convention by default.

The C-runtime side then provides `aether_sleep_ms`, `aether_print`, etc. as thin wrappers over `nanosleep` / `printf` / `exit`. User C code that needs the libc symbols still has them at the original names, untouched.

## Migration

Pure C-side change. No user-Aether-code change required. Existing `extern sleep(...)` declarations in user code (rare — most rely on the built-in) would need to be renamed to `extern aether_sleep_ms(...)`, but those are easy to grep for and the change is mechanical.

## Acceptance

- [ ] No Aether-built-in primitive emits a bare-named C symbol that could collide with libc / POSIX / common third-party libraries.
- [ ] Test added: a probe declares `extern sleep(int)` for libc's sleep and uses both that and the Aether `sleep(...)` built-in in the same file. Builds clean.
- [ ] Documented in `docs/c-interop.md` so anyone writing a C shim knows the contract.

## Notes

- Current PR (`feat/std-http-v2-client`, integration test) works around this by *not* declaring `extern sleep` and relying on the silent built-in. Please file follow-up to remove the workaround once this lands.
- The `pqueue_*` precedent (PR #219, `std/collections/aether_pqueue.c`) is the model for both the rename mechanics and the rationale.

🤖 Filed by Claude Code while building std.http.client v2

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -117,7 +117,17 @@ jobs:
         install: mingw-w64-x86_64-gcc mingw-w64-x86_64-zlib mingw-w64-x86_64-openssl mingw-w64-x86_64-ca-certificates pkg-config make bc
 
     - name: Run CI suite
-      run: make ci
+      # The hardcoded `C:\msys64\...` probe in aether_http.c::load_windows_ca_bundle
+      # doesn't match the GitHub runner's MSYS2 path (it's installed under a
+      # temp dir, not C:\msys64). Export SSL_CERT_FILE explicitly using
+      # cygpath -w so the first probe in get_ssl_ctx() hits regardless of
+      # where MSYS2 lives on this runner. Without this, `tests/regression/
+      # test_https_client.ae` fails with "certificate verify failed".
+      run: |
+        export SSL_CERT_FILE="$(cygpath -w /mingw64/etc/ssl/certs/ca-bundle.crt)"
+        echo "SSL_CERT_FILE=$SSL_CERT_FILE"
+        ls -la "$(cygpath -u "$SSL_CERT_FILE")"
+        make ci
 
   # ============================================================
   # Platform portability — cooperative scheduler, WASM, embedded

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -43,7 +43,14 @@ jobs:
           bc
 
     - name: Run full CI suite
-      run: make ci
+      # See ci.yml::ci-windows for why SSL_CERT_FILE is exported here.
+      # Hardcoded probe paths in aether_http.c don't match the GitHub
+      # runner's MSYS2 install location.
+      run: |
+        export SSL_CERT_FILE="$(cygpath -w /mingw64/etc/ssl/certs/ca-bundle.crt)"
+        echo "SSL_CERT_FILE=$SSL_CERT_FILE"
+        ls -la "$(cygpath -u "$SSL_CERT_FILE")"
+        make ci
 
     - name: Upload Windows binaries
       uses: actions/upload-artifact@v4

--- a/README.md
+++ b/README.md
@@ -194,7 +194,31 @@ make -j8                         # Parallel build
 make help                        # Show all targets
 ```
 
-**Windows:** Use the [release binary](https://github.com/nicolasmd87/aether/releases) — no MSYS2 needed. To build from source, use MSYS2 MinGW 64-bit shell; `make ci` runs the full suite (compiler, ae, stdlib, REPL, C tests, .ae tests, examples) with no skips.
+### Building on Windows
+
+The Aether build is GNU-make based. Use one of the two paths below — `nmake` from a Visual Studio Developer Prompt **will not work** (the Makefile uses GNU-only syntax that NMAKE can't parse).
+
+**Just running Aether? Skip this section** and use the [release binary](https://github.com/nicolasmd87/aether/releases) — no MSYS2 setup required.
+
+**Building from source — recommended (MSYS2 / MinGW-w64):**
+
+1. Install [MSYS2](https://www.msys2.org/) and open the **MSYS2 MinGW 64-bit** shell (not the bare MSYS shell).
+2. Install the toolchain:
+   ```bash
+   pacman -S mingw-w64-x86_64-gcc mingw-w64-x86_64-make \
+             mingw-w64-x86_64-openssl mingw-w64-x86_64-zlib \
+             mingw-w64-x86_64-ca-certificates pkg-config make bc
+   ```
+3. Clone and build:
+   ```bash
+   git clone https://github.com/nicolasmd87/aether.git
+   cd aether
+   make ci   # full suite: compiler, ae, stdlib, REPL, C tests, .ae tests, examples
+   ```
+
+For HTTPS to verify certs, the `mingw-w64-x86_64-ca-certificates` package above provides the bundle at `/mingw64/etc/ssl/certs/ca-bundle.crt`. The runtime auto-detects it; if your install is in a non-standard location, export `SSL_CERT_FILE` to the bundle's Windows path.
+
+**Native MSVC (cl.exe / nmake):** not currently supported as a full build path — tracker [#99](https://github.com/nicolasmd87/aether/issues/99). The MSVC matrix job in CI verifies our public headers parse under `cl.exe` so a future native MSVC port stays feasible, but `make` (the build system itself) requires GNU make. The MSYS2 MinGW build above is the supported source-build path for Windows today.
 
 ## Project Structure
 

--- a/examples/stdlib/http-client.ae
+++ b/examples/stdlib/http-client.ae
@@ -12,9 +12,9 @@ fetch(url: string) {
     body, err = http.get(url)
     if err != "" {
         println("  failed: ${err}")
-        return
+    } else {
+        println("  ok: ${body}")
     }
-    println("  ok: ${body}")
 }
 
 main() {

--- a/tests/integration/tuple_destructure_cross_module/test_tuple_destructure_cross_module.sh
+++ b/tests/integration/tuple_destructure_cross_module/test_tuple_destructure_cross_module.sh
@@ -24,7 +24,12 @@
 #   - infer_function_return_types re-tries when a function's previous
 #     guess was VOID (otherwise an early UNKNOWN-resolved-as-VOID guess
 #     sticks across iterations).
-set -euo pipefail
+# Note: the test harness invokes this via `sh path/to/test.sh`, which
+# resolves to dash on Ubuntu — dash does not implement `-o pipefail`,
+# so this script avoids it. No pipelines are used below; `-eu` alone
+# gives the safety we need (fail-fast on any non-zero, error on
+# unset vars).
+set -eu
 
 DIR="$(cd "$(dirname "$0")" && pwd)"
 AETHER_ROOT="$(cd "$DIR/../../.." && pwd)"

